supporting/shimexe/shim.cs

```diff
@@ -164,5 +183,31 @@ internal struct PROCESS_INFORMATION {
             }
             return config;
         }
+
+        private static string ExpandEnvironmentVariables(IDictionary environment_dictionary, string name)
```

Comment:
[`Environment.ExpandEnvironmentVariables()`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.expandenvironmentvariables) already exists in .NET
How is this implementation different?

Reply:
Existing implementation works with current environment in place and mine expands name using local copy of environment from dictionary. Theoretically, I could go another way, change current environment because we run only one program and then exit, but current approach seems more clear to me. Should I change it?

Reply:
Well, I change it. Actually it is the only case where we spoil current environment. Funny, but I never did it earlier.